src/overcooked_ai_py/mdp/layout_generator.py

```diff
@@ -12,74 +12,156 @@
+        else:
+            # there is no schedule, we are using the same set of mdp_params all the time
+            if mdp_params_schedule_fn is None:
+                assert mdp_params is not None
+                mdp_pg = MDPParamsGenerator(mdp_params_always=mdp_params)
+            else:
+                assert mdp_params is None, "please remove the mdp_params from the variable, " \
+                                           "because mdp_params_schedule_fn exist and we will " \
+                                           "always use the schedule_fn if it exist"
+                mdp_pg = MDPParamsGenerator(params_schedule_fn=mdp_params_schedule_fn)
+            lg = LayoutGenerator(mdp_pg, outer_shape)
+            mdp_fn = lg.generate_padded_mdp
+        return mdp_fn
+
+    def generate_padded_mdp(self, outside_information={}):
         """
-        
+        Return a PADDED MDP with mdp params specified in self.mdp_params
+        """
+        mdp_params = self.mdp_params_generator.generate(outside_information)
+        outer_shape = self.outer_shape
         if "layout_name" in mdp_params.keys() and mdp_params["layout_name"] is not None:
             mdp = OvercookedGridworld.from_layout_name(**mdp_params)
-            mdp_generator_fn = lambda: mdp
-        
-        elif mdp_choices is not None:
-            assert type(mdp_choices) is list
-            
-            # If list of MDPs, randomly choose one at each reset
-            mdp_sizes = []
-            for mdp_name in mdp_choices:
-                mdp = OvercookedGridworld.from_layout_name(mdp_name, **mdp_params)
-                mdp_sizes.append([mdp.width, mdp.height])
-            widths, heights = np.array(mdp_sizes).T
-            min_padding = max(widths), max(heights)            
-            
-            def mdp_generator_fn():
-                chosen_mdp = np.random.choice(mdp_choices)
-                mdp = OvercookedGridworld.from_layout_name(chosen_mdp, **mdp_params)
-                lg = LayoutGenerator(min_padding, mdp_params)
-                mdp_padded = lg.padded_mdp(mdp)
-                return mdp_padded
+            mdp_generator_fn = lambda: self.padded_mdp(mdp)
         else:
-            min_padding = (size_bounds[0][1], size_bounds[1][1])
-            layout_generator = LayoutGenerator(min_padding, mdp_params)
+
+            assert "inner_shape" in mdp_params.keys(), "mdp_params is missing inner_shape"
+            assert "prop_empty" in mdp_params.keys(), "mdp_params is missing prop_empty"
+            assert "prop_feats" in mdp_params.keys(), "mdp_params is missing prop_feats"
+            assert "display" in mdp_params.keys(), "mdp_params is missing display"
+            assert "start_all_orders" in mdp_params.keys(), "mdp_params is missing start_orders"
+            assert "recipe_values" in mdp_params.keys(), "mdp_params is missing recipe_values"
+            assert "recipe_times" in mdp_params.keys(), "mdp_params is missing recipe_times"
```

Comment:
You could replace this with something like
```suggestion
            required_keys = ["inner_shape", "prop_empty", "prop_feats", "display", "start_all_orders", "recipe_values", "recipe_times"]
            missing_keys = [k for k in required_keys if k not in mdp_params.keys()]
            assert len(missing_keys) == 0, "These keys were missing from the mdp_params: {}".format(missing_keys)
```

Reply:
Addressed